client/web/admin/src/components/Settings/UI/CUIBrandingEditor.vue

```diff
@@ -0,0 +1,180 @@
+      />
+    </template>
+  </b-card>
+</template>
+
+<script>
+import CSubmitButton from 'corteza-webapp-admin/src/components/CSubmitButton'
+import { components } from '@cortezaproject/corteza-vue'
+const { CInputColorPicker } = components
+
+export default {
+  name: 'CUIBrandingEditor',
+
+  i18nOptions: {
+    namespaces: 'ui.settings',
+    keyPrefix: 'editor.corteza-studio',
+  },
+
+  components: {
+    CInputColorPicker,
+    CSubmitButton,
+  },
+
+  props: {
+    settings: {
+      type: Object,
+      required: true,
+    },
+
+    processing: {
+      type: Boolean,
+      value: false,
+    },
+
+    success: {
+      type: Boolean,
+      value: false,
+    },
+
+    canManage: {
+      type: Boolean,
+      required: true,
+    },
+  },
+
+  data () {
+    return {
+      colorData: {
+        white: {
+          label: 'White',
+          value: '#FFFFFF',
+        },
+        black: {
+          label: 'Black',
+          value: '#162425',
+        },
+        primary: {
+          label: 'Primary',
+          value: '#0B344E',
+        },
+        secondary: {
+          label: 'Secondary',
+          value: '#758D9B',
+        },
+        success: {
+          label: 'Success',
+          value: '#43AA8B',
+        },
+        warning: {
+          label: 'Warning',
+          value: '#E2A046',
+        },
+        danger: {
+          label: 'Danger',
+          value: '#E54122',
+        },
+        light: {
+          label: 'Light',
+          value: '#E4E9EF',
+        },
+        extraLight: {
+          label: 'Extra Light',
+          value: '#F3F5F7',
+        },
+        dark: {
+          label: 'Dark',
+          value: '#162425',
+        },
+        tertiary: {
+          label: 'Tertiary',
+          value: '#5E727E',
+        },
+        gray200: {
+          label: 'Gray-200',
+          value: '#F9FAFB ',
+        },
+        bodyBg: {
+          label: 'Body Background',
+          value: '#F9FAFB',
+        },
+      },
+      colorTranslations: {
+        modalTitle: this.$t('colorPicker'),
+        saveBtnLabel: this.$t('general:label.saveAndClose'),
+      },
+    }
+  },
+
+  watch: {
+    settings: {
+      immediate: true,
+      handler (settings) {
+        const brandingVariables = settings['ui.branding.variables'] || []
+        if (brandingVariables.length !== 0) {
+          this.colorData.primary.value = this.brandColorValue(brandingVariables, 'primary')
+          this.colorData.secondary.value = this.brandColorValue(brandingVariables, 'secondary')
+        }
+      },
+    },
+  },
+
+  methods: {
+    onSubmit () {
+      const brandVariables = Object.keys(this.colorData).map(key => {
+        return `$${key}: ${this.colorData[key].value}`
+      })
+
+      this.$emit('submit', { 'ui.branding.variables': brandVariables })
+    },
+
+    brandColorValue (brandingVariables, variable) {
+      const brandingVariable = brandingVariables.find(item => item.includes(`$${variable}:`))
```

Comment:
Wouldn't it be easier to define the variable structure, so you can directly access the values?